src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/TraceReporter.kt

```diff
@@ -473,12 +473,14 @@ private val Class<*>.simpleNameForAnonymous: String get() {
 internal fun getObjectNumber(clazz: Class<Any>, obj: Any): Int = objectNumeration
     .computeIfAbsent(clazz) { IdentityHashMap() }
     .computeIfAbsent(obj) { 1 + objectNumeration[clazz]!!.size }
-private fun getObjectNumber(obj: Any): Int = objectNumeration
-    .computeIfAbsent(obj.javaClass) { IdentityHashMap() }
-    .computeIfAbsent(obj) { 1 + objectNumeration[it.javaClass]!!.size }
 
 private val objectNumeration = WeakHashMap<Class<Any>, MutableMap<Any, Int>>()
+    @Synchronized get
```

Comment:
Please use Collections.synchronizedMap instead -- this will protect only reading the hashmap reference, not the hashmap content.